cmd/pbstatuser/main.go

```diff
@@ -20,25 +26,179 @@ import (
+	chAzure      = make(chan Pipeline, ChannelBuffer)
+	chGcp        = make(chan Pipeline, ChannelBuffer)
+	chSend       = make(chan Pipeline, ChannelBuffer)
+	receiverWG   = sync.WaitGroup{}
+	processingWG = sync.WaitGroup{}
+	senderWG     = sync.WaitGroup{}
+)
+
 func processMessage(ctx context.Context, message *kafka.GenericMessage) {
 	logger := ctxval.Logger(ctx)
 	// This is just a test to see if Kafka is configured properly in ephemeral
 	logger.Warn().Msg("RECEIVED KAFKA MESSAGE")
 
-	// TODO: implement the source checking for AWS, Azure, GCP
+	// Get source id
+	asm, err := kafka.NewAvailabilityStatusMessage(message)
+	if err != nil {
+		// TODO: Add error handling
+		return
+	}
+
+	sourceId := asm.SourceID
+
+	// Get sources client
+	sourcesClient, err := clients.GetSourcesClient(ctx)
+	if err != nil {
+		// TODO: Add error handling
+		return
+	}
 
-	// This needs to be done in 4 goroutines.
-	// 1) Reads messages, calls sources to get auth type and ARN and then enqueue the work
-	// in one of three channels (AWS, Azure, GCP)
-	// 2) Reads from AWS channel, performs availability check, has configurable throttling.
-	// 3) Reads from Azure channel, performs availability check, has configurable throttling.
-	// 4) Reads from GCP channel, performs availability check, has configurable throttling.
+	// Fetch authentication from Sources
+	authentication, err := sourcesClient.GetAuthentication(ctx, sourceId)
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			// TODO: Add error handling
+			return
+		}
+		// TODO: Add error handling
+		return
+	}
+
+	switch authentication.ProviderType {
+	case models.ProviderTypeAWS:
+		chAws <- Pipeline{
+			Authentication: authentication,
+			SourceID:       sourceId,
+		}
+	case models.ProviderTypeAzure:
+		chAzure <- Pipeline{
+			Authentication: authentication,
+			SourceID:       sourceId,
+		}
+	case models.ProviderTypeGCP:
+		chGcp <- Pipeline{
+			Authentication: authentication,
+			SourceID:       sourceId,
+		}
+	case models.ProviderTypeNoop:
 
-	// Beware there is no database connection, no job queue, most of what api/worker processes have
-	// is not available because it should not be needed. The worker will simply use SDKs to check
-	// statuses and send out results via kafka.Send function to Sources.
-	// Also make sure that all goroutines are closed correctly on context cancel. A WaitGroup
-	// would be a really good here: https://gist.github.com/fracasula/b579d52daf15426e58aa133d0340ccb0
+	case models.ProviderTypeUnknown:
+		// TODO: Complete
+	}
+}
+
+func checkSourceAvailabilityAzure(ctx context.Context) {
+	defer processingWG.Done()
+
+	for p := range chAzure {
+		// TODO: check if source is avavliable
+		p.Status = StatusUnavailable
+		chSend <- p
+	}
+}
+
+func checkSourceAvailabilityAWS(ctx context.Context) {
+	defer processingWG.Done()
+
+	for p := range chAws {
+		_, err := clients.GetEC2Client(ctx, p.Authentication, "us-east-1")
```

Comment:
I think if you pass empty region it will use the DefaultRegion configuration.

```suggestion
		_, err := clients.GetEC2Client(ctx, p.Authentication, "")
```